Sparkle/SUTouchBar.m

```diff
@@ -0,0 +1,94 @@
+//
+//  SUTouchBar.m
+//  Sparkle
+//
+//  Created by Yuxin Wang on 05/01/2017.
+//  Copyright © 2017 Sparkle Project. All rights reserved.
+//
+
+#import "SUTouchBar.h"
+
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101202
```

Comment:
`NSTouchBar` is marked `NS_CLASS_AVAILABLE_MAC(10_12_2)`, so it will be weakly linked.
To support touch bar with compiling with old SDK, we need a forward declare and `NSClassFromString` trick.